gate-web/src/main/groovy/com/netflix/spinnaker/gate/security/x509/X509AuthenticationUserDetailsService.groovy

```diff
@@ -53,6 +58,22 @@ class X509AuthenticationUserDetailsService implements AuthenticationUserDetailsS
   @Autowired(required = false)
   X509UserIdentifierExtractor userIdentifierExtractor
 
+  @Autowired
+  DynamicConfigService dynamicConfigService
+
+  final ConcurrentMap<String, Long> loginDebounce = new ConcurrentHashMap<>()
```

Comment:
Not a strong feeling ... I'd probably have leaned towards a expire-after-write guava cache but I don't imagine we'll be grief'd by a big map of email address.

Reply:
yeah agreed -  I changed to that